### PR TITLE
Adds GET products/all for privileged users

### PR DESCRIPTION
**Pull request by A-Guldborg**
Some weird merge conflicts arised on #232 after #234 was made. Probably due to me branching out from the wrong branch initially. Here is a new pull request instead :)

This adds an endpoint for the product management with frontend in https://github.com/AnalogIO/shifty-webapp/pull/20 and discussed as endpoint in #217 (and slack).

The API will have the endpoint /api/v2/products/all which gives all products, not restricted to a specific usergroup or restricted to only visible products.

This endpoint is only accessible for users with Board privileges and will allow board members to manage the visibilty of products or products that they otherwise don't have access to themselves in the app.

### CODE REVIEW

File: coffeecard/CoffeeCard.Models/DataTransferObjects/v2/Products/ProductResponse.cs [hunk 1]
```diff
@@ -68,6 +68,14 @@ public class ProductResponse
         [Required]
         public bool IsPerk { get; set; }
 
+        /// <summary>
+        /// Visibility of products for users
+        /// </summary>
+        /// <value>Product visibility</value>
+        /// <example>true</example>
+        [Required]
+        public bool Visible { get; set; }
```
Comment (marfavi) [inline coffeecard/CoffeeCard.Models/DataTransferObjects/v2/Products/ProductResponse.cs:77]: Add this field to example object at the top of this file.

File: coffeecard/CoffeeCard.Tests.Unit/Services/v2/ProductServiceTest.cs [hunk 2]
```diff
@@ -70,16 +70,14 @@ await productService.UpdateProduct(new UpdateProductRequest()
...
 
-            Assert.Collection<UserGroup>(expected,
-                e => e.Equals(UserGroup.Customer),
-                e => e.Equals(UserGroup.Board));
+            Assert.Collection<ProductUserGroup>(result.ProductUserGroup,
+                e => Assert.Equal(UserGroup.Customer, e.UserGroup),
+                e => Assert.Equal(UserGroup.Board, e.UserGroup));
+
+            Assert.DoesNotContain(UserGroup.Barista, result.ProductUserGroup.Select(e => e.UserGroup));
+            Assert.DoesNotContain(UserGroup.Manager, result.ProductUserGroup.Select(e => e.UserGroup));
```
Comment (marfavi) [inline coffeecard/CoffeeCard.Tests.Unit/Services/v2/ProductServiceTest.cs:80]: Doesn't Assert.Collection check that the collection has exactly two elements inside, implying that if it passes, there should be no other elements inside result.ProductUserGroup (making DoesNotContain unnecesary)?

**Review by marfavi — CHANGES_REQUESTED** [review]
Small comments

**Review by marfavi — CHANGES_REQUESTED** [review]
Rename your PR title as well :)

File: coffeecard/CoffeeCard.Tests.Unit/Services/v2/ProductServiceTest.cs (hunk 2 above)
Comment (A-Guldborg) [reply coffeecard/CoffeeCard.Tests.Unit/Services/v2/ProductServiceTest.cs:80]: It does, but I have left it there for a reason:

Testing is one of the strongest ways of documenting expected behaviour. Changes in this area in the future should be very aware that it is not just expected that the collection contains those two elements in `Assert.Collection` - but they should have removed the two elements from `Assert.DoesNotContain`. In other words, I believe they convey important information - despite not strictly being necessary in terms of the code test.

**Review by marfavi — CHANGES_REQUESTED** [review]

**Review by marfavi — APPROVED** [review]
🚀

**Pull request merged**